x-pack/solutions/observability/plugins/slo/server/services/repair_slo.ts

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { type Logger } from '@kbn/core/server';
+import type { IScopedClusterClient } from '@kbn/core/server';
+import type { RepairParams } from '@kbn/slo-schema';
+import pLimit from 'p-limit';
+import { computeHealth, type SLOHealth } from '../domain/services/compute_health';
+import { getSLOTransformId, getSLOSummaryTransformId } from '../../common/constants';
+import type { DefaultTransformManager } from './transform_manager';
+import type { DefaultSummaryTransformManager } from './summay_transform_manager';
+import type { SLODefinition } from '../domain/models/slo';
+import type { SLORepository } from './slo_repository';
+
+interface RepairAction {
+  id: string;
+  revision: number;
+  action: 'recreate-transform' | 'start-transform' | 'stop-transform';
+  transformType: 'rollup' | 'summary';
+  enabled: boolean | undefined;
+}
+
+export class RepairSLO {
+  constructor(
+    private logger: Logger,
+    private scopedClusterClient: IScopedClusterClient,
+    private repository: SLORepository,
+    private transformManager: DefaultTransformManager,
+    private summaryTransformManager: DefaultSummaryTransformManager
+  ) {}
+
+  public async execute(
+    params: RepairParams
+  ): Promise<{ id: string; success: boolean; error?: string }[]> {
+    if (params.list.length > 100) {
+      throw new Error('Cannot repair more than 100 SLOs at once');
+    }
+    const allResults: Array<{ id: string; success: boolean; error?: string }> = [];
+
+    const definitions = await this.repository.findAllByIds(params.list);
+
+    let successCount = 0;
+    let errorCount = 0;
+
+    const headLimiter = pLimit(10);
+
+    await Promise.all(
+      definitions.map(async (definition) => {
+        return headLimiter(async () => {
+          const health = await computeHealth([{ ...definition, instanceId: '*' }], {
```

Comment:
computeHealth takes an array of definitions (you can remove the instanceId) and call computeHealth only once with the definitions retrieved.